packages/go/analysis/ad/adcscache.go

```diff
@@ -27,46 +28,52 @@ import (
 )
 
 type ADCSCache struct {
-	AuthStoreForChainValid          map[graph.ID]cardinality.Duplex[uint32]
-	RootCAForChainValid             map[graph.ID]cardinality.Duplex[uint32]
-	ExpandedCertTemplateControllers map[graph.ID][]uint32
-	CertTemplateEnrollers           map[graph.ID][]*graph.Node // principals that have enrollment on a cert template via `enroll`, `generic all`, `all extended rights` edges
-	CertTemplateControllers         map[graph.ID][]*graph.Node // principals that have privileges on a cert template via `owner`, `generic all`, `write dacl`, `write owner` edges
-	EnterpriseCAEnrollers           map[graph.ID][]*graph.Node // principals that have enrollment rights on an enterprise ca via `enroll` edge
-	PublishedTemplateCache          map[graph.ID][]*graph.Node // cert templates that are published to an enterprise ca
-	HasUPNCertMappingInForest       map[graph.ID]struct{}      // domains where at least one DC in the forest has Schannel UPN cert mapping enabled
-	HasWeakCertBindingInForest      map[graph.ID]struct{}      // domains where at least one DC in the forest has Kerberos weak cert binding enabled
+	sync.RWMutex
```

Comment:
I have two concerns about this particular change:

This promotes an ambiguous call surface to a consumer of the `ADCSCache` type. The embedding will elevate all exported receiver functions defined in `sync.RWMutex` to the available call surface for `ADCSCache`.

This hides some important semantics. A mutex in golang comes with some usage caveats outlined in https://go.dev/src/sync/mutex.go namely: `A Mutex must not be copied after first use.`. While the current implementation correctly references the `ADCSCache` type by reference `s *ADCSCache` it is not immediately obvious to users of this type that it now has semantic restrictions on when it can be copied. 

My suggestion here is a refactor that sequesters the mutex as a non-exported struct member.

Reply:
👍 Thanks for explaining, ❤️ the added protection this buys us.